website/content/docs/ui/custom-messages.mdx

```diff
@@ -0,0 +1,147 @@
+
+1. **Delete old messages**. Vault supports a maximum of 100 messages per namespace at a
+   time. Practice good message hygiene by regularly deleting expired and outdated
+   messages.
+
+</Tip>
+
+## Before you start
+
+- **You must have Vault Enterprise 1.16.0 or higher installed.**
+- **You must have the appropriate permissions**:
+  - You must have `list` permission for the `sys/config/ui/custom-messages` endpoint.
+  - To **create messages**, you must have `read` permission for the `sys/config/ui/custom-messages/:id` endpoint and `create` permission for the `sys/config/ui/custom-messages` endpoint.
+  - To **edit messages**, you must have `read` and `update` permission for the `sys/config/ui/custom-messages/:id` endpoint.
+  - To **delete messages**, `delete` permission for the `sys/config/ui/custom-messages/:id` endpoint.
+
+## Add a custom message
+
+1. Navigate to the **Settings** section in the Vault UI sidebar and select **Custom
+   Messages**.
+
+1. On the **Custom messages** page, select whether you want the message
+   to appear on the Vault UI login page or after a user logs in.
+
+1. Select the **+ Create message** button in the toolbar to open the **Create message** form.
+
+1. On the **Create message** form:
+
+   - Select the locations where your message should appear.
+   - Select the message type.
+   - Provide a title and the message text. For important messages, we recommend
+     keeping the text short and including a link to more information rather
+     than writing a longer message that users may not take the time to read.
+   - Set a start time when your message will publish. By default, messages
+     publish at midnight in your local timezone.
+   - Set an optional end time for the message. By default, messages do not
+     expire.
+
+1. Click **Preview** to see how your message will appear to users.
+
+1. Click **Create message** to save your new message.
+
+### Namespaces consideration
+
+Child [namespaces](/vault/docs/enterprise/namespaces) inherit all messages
+created on the parent namespace. For example, your cluster has the following
+namespace hierarchy.
+
+<CodeBlockConfig hideClipboard>
+
+```plaintext
+─ admin
+    ├── finance
+    └── marketing
+        ├── digital-marketing
+        └── events
+```
+
+</CodeBlockConfig>
+
+Custom messages created on the `admin` namespace apply to all child namespaces:
+`fiance`, `marketing`, `marketing/digital-marketing` and `marketing/events`.
+
+To create a custom message intended to the marketing team but not for `finance`,
+you must log into the `marketing` namespace to create a message.
+
+1. On the Vault UI login page, enter `marketing` in the **Namespace** text
+   field, and select an authentication method to log in.
+
+1. Select **Custom Messages**.
+
+1. Select the **+ Create message** button in the toolbar to open and fill out
+   the **Create message** form.
+
+   ![Custom message](/img/ui-custom-msg.png)
+
```

Comment:
```suggestion
```

The screenshot demonstrates what a custom message looks like for a user who has logged in under the target namespace, not the create UI, so it belongs at the end when you discuss the results, not in the middle of the admin instructions.